### PR TITLE
[Feature] K-Net mask cost

### PR DESCRIPTION
**Pull request by chhluo**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

1. Add `MaskCost` for K-Net.
2. Update `panoptic_gt_processing`
3. Update docstring in mask_hungarian_assigner
4. Update `MaskFormerHead.preprocess_gt`.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
5. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
6. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
7. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdet/core/bbox/match_costs/match_cost.py [hunk 1]
```diff
@@ -357,3 +357,42 @@ def __call__(self, cls_pred, gt_labels):
...
 
         return cls_cost * self.weight
+
+
+@MATCH_COST.register_module()
+class MaskCost:
+    """MaskCost.
+
+    Args:
+        weight (int | float, optional): loss_weight
```
Comment (BIGWangYuDong) [inline mmdet/core/bbox/match_costs/match_cost.py:367]: optional here is strange, `weight` need be float or int. Add defaluts value. same as `use_sigmoid`

File: mmdet/models/utils/panoptic_gt_processing.py [hunk 2]
```diff
@@ -2,10 +2,14 @@
...
 
-def preprocess_panoptic_gt(gt_labels, gt_masks, gt_semantic_seg, num_things,
-                           num_stuff, img_metas):
+def preprocess_panoptic_gt(gt_labels,
+                           gt_masks,
+                           gt_semantic_seg,
+                           num_things,
+                           num_stuff,
+                           img_metas,
+                           return_seperate=False):
```
Comment (BIGWangYuDong) [inline mmdet/models/utils/panoptic_gt_processing.py:11]: doc strings need be updated

File: mmdet/models/utils/panoptic_gt_processing.py [hunk 3]
```diff
@@ -51,18 +57,28 @@ def preprocess_panoptic_gt(gt_labels, gt_masks, gt_semantic_seg, num_things,
...
+    if return_seperate:
+        if len(stuff_labels_list) > 0:
+            stuff_masks = torch.stack(stuff_masks_list, dim=0)
+            stuff_labels = torch.stack(stuff_labels_list, dim=0)
+        else:
+            stuff_masks = gt_semantic_seg.new_zeros(
+                size=(0, ) + gt_semantic_seg.shape[-2:])
+            stuff_labels = gt_semantic_seg.new_zeros(size=(0, ))
+
+        return things_labels, things_masks, stuff_labels, stuff_masks
```
Comment (BIGWangYuDong) [inline mmdet/models/utils/panoptic_gt_processing.py:73]: the return is different, may occur some potential error. Maybe we can add `None` to force set return value that not needed when return_seperate is False
